### PR TITLE
Remove `Simulate` from dependencies

### PR DESCRIPTION
**Pull request by sivkoff**
**What**:
Removed `react-tom/test-utils` as a dependency

**Why**:
The biggest reason is #129 (see more details #206)

**How**:
By creating the `createReactEventSimulator` function that makes possible to simulate React events

**Checklist**:

- [x] Documentation N/A
- [x] Tests N/A
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by sivkoff** [comment]
@kentcdodds I've made rebase, but I'm concerned if I have broken something 😅

File: src/__tests__/events.js [hunk 1]
```diff
@@ -156,10 +156,12 @@ eventTypes.forEach(({type, events, elementType, init}) => {
...
-  expect(handleChange).toHaveBeenCalledTimes(1)
+  const spy = jest.fn()
+
+  const {container} = render(<input onChange={spy} />)
+  const input = container.firstChild
+  fireEvent.change(input, {target: {value: 'c'}})
+  expect(spy).toHaveBeenCalledTimes(1)
 })
+
+/* eslint complexity:0 */
```
Comment (kentcdodds) [inline src/__tests__/events.js]: Why were these changes necessary? It looks like a pure refactor.

File: src/index.js [hunk 2]
```diff
@@ -59,14 +58,59 @@ function cleanupAtContainer(container) {
...
+  return simulate({
+    target: node,
+    type: eventName,
+    ...init,
+  })
+}
+
+// function written after some investigation here:
+// https://github.com/facebook/react/issues/10135#issuecomment-401496776
+function setNativeValue(element, value) {
```
Comment (kentcdodds) [inline src/index.js]: Maybe we should export this function from `dom-testing-library` so we can reuse that code here 🤔 

Would you like to make that PR there first? Should be a pretty simple one :)

File: src/index.js [hunk 3]
```diff
@@ -59,14 +58,59 @@ function cleanupAtContainer(container) {
   mountedContainers.delete(container)
 }
 
+const originalChange = fireEvent.change
```
Comment (kentcdodds) [inline src/index.js]: Is it necessary to override the `change` event for this to work? I think we can remove the `change` event code here... I think that one works as-is

File: src/index.js [hunk 4]
```diff
@@ -59,14 +58,59 @@ function cleanupAtContainer(container) {
...
+    setNativeValue(node, init.target.value)
+  }
+  return originalChange(node, init)
+}
+
+// probably there is a better way to get access to a component instance
+const [
+  getInstanceFromNode,
+] = ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.Events
+const createReactEventSimulator = eventName => (node, init) => {
```
Comment (kentcdodds) [inline src/index.js]: Calling the prop directly sorta works, but I _think_ I'd rather come up with a way to fire the browser events that react is looking for when it triggers the callback.

File: src/__tests__/events.js (hunk 1 above)
Comment (sivkoff) [reply src/__tests__/events.js]: It's actually from your commit to original `pr/remove-simulate`, but agree we can drop these changes.

File: src/index.js (hunk 3 above)
Comment (sivkoff) [reply src/index.js]: The same as above, but I can check if it's necessary.

File: src/index.js (hunk 2 above)
Comment (sivkoff) [reply src/index.js]: Good idea! :)

File: src/index.js (hunk 4 above)
Comment (sivkoff) [reply src/index.js]: OK, I'll dig into that deeper 😄

**Comment by sivkoff** [comment]
Sure, I will do 👍

File: src/index.js (hunk 2 above)
Comment (sivkoff) [reply src/index.js]: Added export in the `dom-testing-library`:
https://github.com/kentcdodds/dom-testing-library/pull/137

**Comment by sivkoff** [comment]
OK, I got tests passing with using just native events 😄(thanks to @weyert's [comment](https://github.com/kentcdodds/react-testing-library/issues/206#issuecomment-432872702))
I'll commit it after some refactoring.

**Comment by weyert** [comment]
Awesome!

**Comment by kentcdodds** [comment]
I'm struggling to release this. There's something wrong with our automation. I may not get to it until after next week.

**Comment by weyert** [comment]
The other ticket suggests it's a rights issue?

**Comment by kentcdodds** [comment]
Sweet! Got this working!
